pyprof/nvtx/nvmarker.py

```diff
@@ -312,15 +400,26 @@ def wrapper_func(*args, **kwargs):
             # Pop trace marker
             nvtx.range_pop()
 
+        if config.capture_input_ops:
+            capture_outputs(call_id, result)
+            # Store the callid -> op_name mapping
+            if config.func_stack_enabled:
+                call_id_to_op_map[call_id] = traceMarker_dict['funcStack']
+                call_id = call_id + 1
+
         return result
 
     setattr(mod, fn_name, wrapper_func)
 
 
-def argMarker(mod, op, args, kwargs):
+def argMarker(mod, op, args, kwargs, idx=-1, inputid_list=[]):
     #For this function args is a tuple and kwargs is a dict
+    global call_id_to_op_map
```

Comment:
couldn't you get away from globals by using functions to read/write the structures?

Reply:
Will that ticket be done anytime soon? Seems like the type of ticket that will keep getting delayed for months.

Aditya mentioned a singleton.  Could use that short term just to wrap the globals?

Reply:
Its in my sprint for this week so that is the intention. 
A singleton for the input ops recording and function stack work is the solution I have in mind.

Reply:
Which ticket? I see 64 in your sprint, not 69.

Reply:
As I understood it, those are the same thing. I need to refactor the code and spend some time profiling it.

Reply:
Two very different tasks. One is profiling and optimizing the code. The other is moving code around into different files for organization.

Is it a lot of work to put these global variables into a singleton? Aditya has expressed desire not to have globals, and I don't feel comfortable just pushing it off to a refactor task.

Reply:
It was none trivial but I think i got it into a singleton.
The charter for the refactor work was to trim dlprof code out of nvmarker as well as profiling to see if there was an issue we introduced.